### ISSUE TITLE
upgrade from nodejs-14.19.0 to nodejs-18.12.1 appears to break BGPalerter

### ISSUE THREAD

**Issue by ccaputo** [nttgin/BGPalerter#1030]
With nodejs-14.19.0 I was able to run BGPalerter-1.31.0 without problems.

With nodejs-18.12.1 and the BGPalerter-1.31.0 folder built with `npm install` from nodejs-14.19.0 I get:

```
> bgpalerter@1.31.0 serve
> babel-node index.js

Loaded config: /home/homedir/BGPalerter/config.yml
BGPalerter, version: 1.31.0 environment: production
/home/homedir/BGPalerter/src/worker.js:16
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if (i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s() { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         ^

TypeError: Cannot set property closed of #<Readable> which has only a getter
    at patch (/home/homedir/BGPalerter/node_modules/restify/lib/request.js:848:30)
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/restify/lib/server.js:33:1)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)

Node.js v18.12.1

```

With nodejs-18.12.1 and a fresh BGPalerter-1.31.0 folder built with `npm install` from nodejs-18.12.1 I get:

```
> bgpalerter@1.31.0 serve                                                                                                                                                                                                                                                                                                                                            
> babel-node index.js                                                                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                                                                     
Loaded config: /home/homedir/BGPalerter/config.yml
BGPalerter, version: 1.31.0 environment: production
/home/homedir/BGPalerter/src/worker.js:16                                                                                                                                                                                                                                                                                                                            
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if 
(i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s(
) { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }
                                                                                                                                                                                  
                                                                                         
                                                                                                                                                                                                                                                                               ^
                                                                                         
TypeError: Cannot set property closed of #<Readable> which has only a getter                                                                                                      
    at patch (/home/homedir/BGPalerter/node_modules/restify/lib/request.js:848:30)
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/restify/lib/server.js:33:1)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/restify/lib/index.js:10:14)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/home/homedir/BGPalerter/src/utils/restApi.js:1:1)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/home/homedir/BGPalerter/src/processMonitors/uptimeApi.js:35:1)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Worker.main (/home/homedir/BGPalerter/src/worker.js:99:38)
    at new Worker (/home/homedir/BGPalerter/src/worker.js:56:18)
    at Object.<anonymous> (/home/homedir/BGPalerter/index.js:200:26)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)
    at Function.runMain (node:internal/modules/run_main:81:12)
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/@babel/node/src/_babel-node.ts:215:12)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1213:10)

Node.js v18.12.1
```

With nodejs-18.12.1 and a fresh BGPalerter-1.31.1 folder built with `npm install` from nodejs-18.12.1 I get:

```
> bgpalerter@1.31.1 serve                                                                                                                                                                                                                                                                                                                                            
> babel-node index.js                                                                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                                                                     
Loaded config: /home/homedir/BGPalerter/config.yml                                                                                                                                                                                                                                                                                                                   
BGPalerter, version: 1.31.1 environment: production                                                                                                                                                                                                                                                                                                                  
/home/homedir/BGPalerter/src/worker.js:12                                                                                                                                                                                                                                                                                                                            
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if 
(i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s(
) { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }                                                                   
                                                                                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                               ^                                                                                     
                                                                                                                                                                                                                                                                                                                                                                     
TypeError: Cannot set property closed of #<Readable> which has only a getter                                                                                                                                                                                                                                                                                         
    at patch (/home/homedir/BGPalerter/node_modules/restify/lib/request.js:848:30)                                                                                                                                                                                                                                                                                   
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/restify/lib/server.js:33:1)                                                                                                                                                                                                                                                                         
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/restify/lib/index.js:10:14)                                                                                                                                                                                                                                                                         
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Object.<anonymous> (/home/homedir/BGPalerter/src/utils/restApi.js:1:1)                                                                                                                                                                                                                                                                                        
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Object.<anonymous> (/home/homedir/BGPalerter/src/processMonitors/uptimeApi.js:35:1)                                                                                                                                                                                                                                                                           
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Worker.main (/home/homedir/BGPalerter/src/worker.js:99:38)                                                                                                                                                                                                                                                                                                    
    at new Worker (/home/homedir/BGPalerter/src/worker.js:56:18)                                                                                                                                                                                                                                                                                                     
    at Object.<anonymous> (/home/homedir/BGPalerter/index.js:200:26)                                                                                                                                                                                                                                                                                                 
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Function.runMain (node:internal/modules/run_main:81:12)                                                                                                                                                                                                                                                                                                       
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/@babel/node/src/_babel-node.ts:223:12)                                                                                                                                                                                                                                                              
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
                                                                                         
Node.js v18.12.1                                   
```

With nodejs-18.12.1 and a development/pre-release BGPalerter-1.31.2 folder built with `npm install` from nodejs-18.12.1 I get:

```
> bgpalerter@1.31.2 serve                                                                                                                                                                                                                                                                                                                                            
> babel-node index.js                                                                                                                                                                                                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                                                                     
Loaded config: /home/homedir/BGPalerter/config.yml                                                                                                                                                                                                                                                                                                                   
BGPalerter, version: 1.31.2 environment: production                                                                                                                                                                                                                                                                                                                  
/home/homedir/BGPalerter/src/worker.js:12                                                                                                                                                                                                                                                                                                                            
function _createForOfIteratorHelper(o, allowArrayLike) { var it = typeof Symbol !== "undefined" && o[Symbol.iterator] || o["@@iterator"]; if (!it) { if (Array.isArray(o) || (it = _unsupportedIterableToArray(o)) || allowArrayLike && o && typeof o.length === "number") { if (it) o = it; var i = 0; var F = function F() {}; return { s: F, n: function n() { if 
(i >= o.length) return { done: true }; return { done: false, value: o[i++] }; }, e: function e(_e) { throw _e; }, f: F }; } throw new TypeError("Invalid attempt to iterate non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); } var normalCompletion = true, didErr = false, err; return { s: function s(
) { it = it.call(o); }, n: function n() { var step = it.next(); normalCompletion = step.done; return step; }, e: function e(_e2) { didErr = true; err = _e2; }, f: function f() { try { if (!normalCompletion && it["return"] != null) it["return"](); } finally { if (didErr) throw err; } } }; }                                                                   
                                                                                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                                               ^                                                                                     
                                                                                                                                                                                                                                                                                                                                                                     
TypeError: Cannot set property closed of #<Readable> which has only a getter                                                                                                                                                                                                                                                                                         
    at patch (/home/homedir/BGPalerter/node_modules/restify/lib/request.js:848:30)                                                                                                                                                                                                                                                                                   
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/restify/lib/server.js:33:1)                                                                                                                                                                                                                                                                         
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/restify/lib/index.js:10:14)                                                                                                                                                                                                                                                                         
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Object.<anonymous> (/home/homedir/BGPalerter/src/utils/restApi.js:1:1)                                                                                                                                                                                                                                                                                        
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Object.<anonymous> (/home/homedir/BGPalerter/src/processMonitors/uptimeApi.js:35:1)                                                                                                                                                                                                                                                                           
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Module.require (node:internal/modules/cjs/loader:1061:19)                                                                                                                                                                                                                                                                                                     
    at require (node:internal/modules/cjs/helpers:103:18)                                                                                                                                                                                                                                                                                                            
    at Worker.main (/home/homedir/BGPalerter/src/worker.js:99:38)                                                                                                                                                                                                                                                                                                    
    at new Worker (/home/homedir/BGPalerter/src/worker.js:56:18)                                                                                                                                                                                                                                                                                                     
    at Object.<anonymous> (/home/homedir/BGPalerter/index.js:200:26)                                                                                                                                                                                                                                                                                                 
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Module._compile (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:136:24)                                                                                                                                                                                                                                                                           
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                             
    at Object.newLoader [as .js] (/home/homedir/BGPalerter/node_modules/pirates/lib/index.js:141:7)                                                                                                                                                                                                                                                                  
    at Module.load (node:internal/modules/cjs/loader:1037:32)                                                                                                                                                                                                                                                                                                        
    at Function.Module._load (node:internal/modules/cjs/loader:878:12)                                                                                                                                                                                                                                                                                               
    at Function.runMain (node:internal/modules/run_main:81:12)                                                                                                                                                                                                                                                                                                       
    at Object.<anonymous> (/home/homedir/BGPalerter/node_modules/@babel/node/src/_babel-node.ts:223:12)                                                                                                                                                                                                                                                              
    at Module._compile (node:internal/modules/cjs/loader:1159:14)                                                                                                                                                                                                                                                                                                    
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1213:10)                                                                                                                                                                                                                                                                                      
                                                                                                                                                                                                                                                                                                                                                                     
Node.js v18.12.1                                                                                                                                                                                                                                                                                                                                                     
```

Thank you for any advice.

**Comment by massimocandela** [comment]
Hi @ccaputo,

The code is able to run on node 18 without modifications.
However, I need a few weeks to test the migration.

I'm not quite sure about the error you get. I would do:
```
git reset --hard
git checkout dev
git pull
npm ci
npm run serve
```
Don't do [npm install, do npm ci instead](https://betterprogramming.pub/npm-ci-vs-npm-install-which-should-you-use-in-your-node-js-projects-51e07cb71e26)

Also, I pushed [here](https://github.com/nttgin/BGPalerter/tree/node18) the complete migration (which includes changes to some config/build files). You can try with that.

**Comment by massimocandela** [comment]
The gitlab CI was triggering the same issue, while other vms I'm using don't
Apparently it is related to restify, which I now updated. The new restify version contains breaking changes, which I believe they are not affecting us (and the tests are ok). However, don't consider this production ready yet.

**Comment by ccaputo** [comment]
Understood re not production.

In a fresh dir I did `git checkout node18`. Version e6ebb35fc6a073ff9125854eda38f8f6418f02cf. Then `npm clean-install` and am now running it. Possible messages of interest:

```
> bgpalerter@1.31.1 serve
> babel-node index.js

Loaded config: /home/homedir/BGPalerter/config.yml
BGPalerter, version: 1.31.1 environment: production
(node:6774) [DEP0111] DeprecationWarning: Access to process.binding('http_parser') is deprecated.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

Thanks!

**Comment by massimocandela** [comment]
Hi @ccaputo,

The warning is raised by restify. There is currently an issue open in restify's repo. I forked and submitted a PR that should address the issue. In the branch `node18` I updated the dependency to use my forked version (while waiting for the PR to be merged), which runs fine.

I will now package another pre-release of BGPalerter and will run it for some time on our systems with Node18. Please, report if you find any other issue.

**Comment by ccaputo** [comment]
> Pre-release [v1.32.0](https://github.com/nttgin/BGPalerter/releases/tag/v1.32.0)

This pre-release (specifically 2b61b6a98bfd7751385fa18ad280dbf6526e4156) appears to be working well for me. No warnings. Clean daemon.log and error.log.

**Comment by massimocandela** [comment]
released in [v1.32.0](https://github.com/nttgin/BGPalerter/releases/tag/v1.32.0)
